gammapy/detect/cwt.py

```diff
@@ -108,34 +139,48 @@ class CWT(object):
         - Fill ``self.model += res * (res > 0)``
         - What is this??
 
-
     - max_scale_image : 2D image, value = index of scale where emission is most significant
 
-    Parameters
+    Examples
+    --------
+    TODO
+
+    References
     ----------
-    min_scale : float
-        first scale used
-    nscales : int
-        number of scales considered
-    scale_step : float
-        base scaling factor
+    R. Terrier, L. Demanet, I.A. Grenier, and J.P. Antoine. Wavelet
+    analysis of EGRET data. Universite   Paris VII & Sap CEA Saclay.
+    Institut de physique theorique et mathematique, Universite
+    catholique de Louvain, Louvain-la-Neuve
     """
 
-    def __init__(self, min_scale, nscales, scale_step):
+    def __init__(self, min_scale, nscales, scale_step, max_iter=10, tol=1e-5):
         self.kernbase = dict()
+
         self.scales = dict()
         self.nscale = nscales
         self.scale_step = scale_step
+
         for idx_scale in np.arange(0, nscales):
             scale = min_scale * scale_step ** idx_scale
             self.scales[idx_scale] = scale
             self.kernbase[idx_scale] = difference_of_gauss_kernel(scale, scale_step)
+        self.max_iter=max_iter
+        self.tol=tol
 
         max_scale = min_scale * scale_step ** nscales
         self.kern_approx = gauss_kernel(max_scale)
 
     def set_data(self, counts, background):
-        """Set input images."""
+        """
+        Set input images.
+
+        TODO : RENAME.
+        Like fit methods in sklearn. I like that.
```

Comment:
I'm not sure [SkyImageCollection](http://docs.gammapy.org/en/latest/api/gammapy.image.SkyImageCollection.html) is a good idea. The only methods it has are `read`, `write` and `info`. And for this application, we probably want to have our own `read`, `write` and `info` that's tailored to our application (e.g. how to group HDUs from the iterations in the FITS file, or in the info printout show the most useful numbers for this algorithm.
My gut feeling is that a `CWTData` class that doesn't sub-class `SkyImageCollection` would work nicely here and that would be the first thing I try.

But as I said, there's many ways to structure the code ... let's see what @OlgaVorokh proposes.